java/arcs/core/allocator/Allocator.kt

```diff
@@ -77,6 +77,11 @@ class Allocator private constructor(
      * Start a new Arc given a [Plan] and return the generated [ArcId].
      */
     suspend fun startArcForPlan(arcName: String, plan: Plan): ArcId {
+        if (plan.arcId !== null) {
+            if (!readPartitions(plan.arcId!!.toArcId()).isEmpty()) {
+                return plan.arcId!!.toArcId()
+            }
+        }
```

Comment:
You can get rid of the extra null checks by using `?.let`:

```
plan.arcId?.let { arcId ->
    if (!readPartitions(arcId.toArcId()).isEmpty()) {
        return arcId.toArcId()
    }
}
```

Reply:
Even simpler: 

```
plan.arcId?.toArcId()?.let { arcId ->
    if (!readPartitions(arcId).isEmpty()) {
        return arcId
    }
}
```

Reply:
this doesn't seem better, if anything it's less clear what's going on.